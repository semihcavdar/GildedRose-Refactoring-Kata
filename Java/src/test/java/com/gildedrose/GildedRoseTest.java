package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

    @Test
    void foo() {
        CombinationApprovals.verifyAllCombinations(this::doUpdate,
            new String[]{"foo","Aged Brie","Backstage passes to a TAFKAL80ETC concert","Sulfuras, Hand of Ragnaros"},
            new Integer[]{0,-1,3,17,20,50},
            new Integer[]{0,-1,1,10,30,48,49,60,100});
    }

    private String doUpdate(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
