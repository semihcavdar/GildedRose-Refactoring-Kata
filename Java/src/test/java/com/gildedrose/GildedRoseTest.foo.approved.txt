[foo, 0, 0] => foo, -1, 0 
[foo, 0, -1] => foo, -1, -1 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 10] => foo, -1, 8 
[foo, 0, 30] => foo, -1, 28 
[foo, 0, 48] => foo, -1, 46 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 60] => foo, -1, 58 
[foo, 0, 100] => foo, -1, 98 
[foo, -1, 0] => foo, -2, 0 
[foo, -1, -1] => foo, -2, -1 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 10] => foo, -2, 8 
[foo, -1, 30] => foo, -2, 28 
[foo, -1, 48] => foo, -2, 46 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 60] => foo, -2, 58 
[foo, -1, 100] => foo, -2, 98 
[foo, 3, 0] => foo, 2, 0 
[foo, 3, -1] => foo, 2, -1 
[foo, 3, 1] => foo, 2, 0 
[foo, 3, 10] => foo, 2, 9 
[foo, 3, 30] => foo, 2, 29 
[foo, 3, 48] => foo, 2, 47 
[foo, 3, 49] => foo, 2, 48 
[foo, 3, 60] => foo, 2, 59 
[foo, 3, 100] => foo, 2, 99 
[foo, 17, 0] => foo, 16, 0 
[foo, 17, -1] => foo, 16, -1 
[foo, 17, 1] => foo, 16, 0 
[foo, 17, 10] => foo, 16, 9 
[foo, 17, 30] => foo, 16, 29 
[foo, 17, 48] => foo, 16, 47 
[foo, 17, 49] => foo, 16, 48 
[foo, 17, 60] => foo, 16, 59 
[foo, 17, 100] => foo, 16, 99 
[foo, 20, 0] => foo, 19, 0 
[foo, 20, -1] => foo, 19, -1 
[foo, 20, 1] => foo, 19, 0 
[foo, 20, 10] => foo, 19, 9 
[foo, 20, 30] => foo, 19, 29 
[foo, 20, 48] => foo, 19, 47 
[foo, 20, 49] => foo, 19, 48 
[foo, 20, 60] => foo, 19, 59 
[foo, 20, 100] => foo, 19, 99 
[foo, 50, 0] => foo, 49, 0 
[foo, 50, -1] => foo, 49, -1 
[foo, 50, 1] => foo, 49, 0 
[foo, 50, 10] => foo, 49, 9 
[foo, 50, 30] => foo, 49, 29 
[foo, 50, 48] => foo, 49, 47 
[foo, 50, 49] => foo, 49, 48 
[foo, 50, 60] => foo, 49, 59 
[foo, 50, 100] => foo, 49, 99 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, -1] => Aged Brie, -1, 1 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 10] => Aged Brie, -1, 12 
[Aged Brie, 0, 30] => Aged Brie, -1, 32 
[Aged Brie, 0, 48] => Aged Brie, -1, 50 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 60] => Aged Brie, -1, 60 
[Aged Brie, 0, 100] => Aged Brie, -1, 100 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, -1] => Aged Brie, -2, 1 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 10] => Aged Brie, -2, 12 
[Aged Brie, -1, 30] => Aged Brie, -2, 32 
[Aged Brie, -1, 48] => Aged Brie, -2, 50 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 60] => Aged Brie, -2, 60 
[Aged Brie, -1, 100] => Aged Brie, -2, 100 
[Aged Brie, 3, 0] => Aged Brie, 2, 1 
[Aged Brie, 3, -1] => Aged Brie, 2, 0 
[Aged Brie, 3, 1] => Aged Brie, 2, 2 
[Aged Brie, 3, 10] => Aged Brie, 2, 11 
[Aged Brie, 3, 30] => Aged Brie, 2, 31 
[Aged Brie, 3, 48] => Aged Brie, 2, 49 
[Aged Brie, 3, 49] => Aged Brie, 2, 50 
[Aged Brie, 3, 60] => Aged Brie, 2, 60 
[Aged Brie, 3, 100] => Aged Brie, 2, 100 
[Aged Brie, 17, 0] => Aged Brie, 16, 1 
[Aged Brie, 17, -1] => Aged Brie, 16, 0 
[Aged Brie, 17, 1] => Aged Brie, 16, 2 
[Aged Brie, 17, 10] => Aged Brie, 16, 11 
[Aged Brie, 17, 30] => Aged Brie, 16, 31 
[Aged Brie, 17, 48] => Aged Brie, 16, 49 
[Aged Brie, 17, 49] => Aged Brie, 16, 50 
[Aged Brie, 17, 60] => Aged Brie, 16, 60 
[Aged Brie, 17, 100] => Aged Brie, 16, 100 
[Aged Brie, 20, 0] => Aged Brie, 19, 1 
[Aged Brie, 20, -1] => Aged Brie, 19, 0 
[Aged Brie, 20, 1] => Aged Brie, 19, 2 
[Aged Brie, 20, 10] => Aged Brie, 19, 11 
[Aged Brie, 20, 30] => Aged Brie, 19, 31 
[Aged Brie, 20, 48] => Aged Brie, 19, 49 
[Aged Brie, 20, 49] => Aged Brie, 19, 50 
[Aged Brie, 20, 60] => Aged Brie, 19, 60 
[Aged Brie, 20, 100] => Aged Brie, 19, 100 
[Aged Brie, 50, 0] => Aged Brie, 49, 1 
[Aged Brie, 50, -1] => Aged Brie, 49, 0 
[Aged Brie, 50, 1] => Aged Brie, 49, 2 
[Aged Brie, 50, 10] => Aged Brie, 49, 11 
[Aged Brie, 50, 30] => Aged Brie, 49, 31 
[Aged Brie, 50, 48] => Aged Brie, 49, 49 
[Aged Brie, 50, 49] => Aged Brie, 49, 50 
[Aged Brie, 50, 60] => Aged Brie, 49, 60 
[Aged Brie, 50, 100] => Aged Brie, 49, 100 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 30] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 48] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 60] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 100] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, -1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 10] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 30] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 48] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 60] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 100] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 3, 0] => Backstage passes to a TAFKAL80ETC concert, 2, 3 
[Backstage passes to a TAFKAL80ETC concert, 3, -1] => Backstage passes to a TAFKAL80ETC concert, 2, 2 
[Backstage passes to a TAFKAL80ETC concert, 3, 1] => Backstage passes to a TAFKAL80ETC concert, 2, 4 
[Backstage passes to a TAFKAL80ETC concert, 3, 10] => Backstage passes to a TAFKAL80ETC concert, 2, 13 
[Backstage passes to a TAFKAL80ETC concert, 3, 30] => Backstage passes to a TAFKAL80ETC concert, 2, 33 
[Backstage passes to a TAFKAL80ETC concert, 3, 48] => Backstage passes to a TAFKAL80ETC concert, 2, 50 
[Backstage passes to a TAFKAL80ETC concert, 3, 49] => Backstage passes to a TAFKAL80ETC concert, 2, 50 
[Backstage passes to a TAFKAL80ETC concert, 3, 60] => Backstage passes to a TAFKAL80ETC concert, 2, 60 
[Backstage passes to a TAFKAL80ETC concert, 3, 100] => Backstage passes to a TAFKAL80ETC concert, 2, 100 
[Backstage passes to a TAFKAL80ETC concert, 17, 0] => Backstage passes to a TAFKAL80ETC concert, 16, 1 
[Backstage passes to a TAFKAL80ETC concert, 17, -1] => Backstage passes to a TAFKAL80ETC concert, 16, 0 
[Backstage passes to a TAFKAL80ETC concert, 17, 1] => Backstage passes to a TAFKAL80ETC concert, 16, 2 
[Backstage passes to a TAFKAL80ETC concert, 17, 10] => Backstage passes to a TAFKAL80ETC concert, 16, 11 
[Backstage passes to a TAFKAL80ETC concert, 17, 30] => Backstage passes to a TAFKAL80ETC concert, 16, 31 
[Backstage passes to a TAFKAL80ETC concert, 17, 48] => Backstage passes to a TAFKAL80ETC concert, 16, 49 
[Backstage passes to a TAFKAL80ETC concert, 17, 49] => Backstage passes to a TAFKAL80ETC concert, 16, 50 
[Backstage passes to a TAFKAL80ETC concert, 17, 60] => Backstage passes to a TAFKAL80ETC concert, 16, 60 
[Backstage passes to a TAFKAL80ETC concert, 17, 100] => Backstage passes to a TAFKAL80ETC concert, 16, 100 
[Backstage passes to a TAFKAL80ETC concert, 20, 0] => Backstage passes to a TAFKAL80ETC concert, 19, 1 
[Backstage passes to a TAFKAL80ETC concert, 20, -1] => Backstage passes to a TAFKAL80ETC concert, 19, 0 
[Backstage passes to a TAFKAL80ETC concert, 20, 1] => Backstage passes to a TAFKAL80ETC concert, 19, 2 
[Backstage passes to a TAFKAL80ETC concert, 20, 10] => Backstage passes to a TAFKAL80ETC concert, 19, 11 
[Backstage passes to a TAFKAL80ETC concert, 20, 30] => Backstage passes to a TAFKAL80ETC concert, 19, 31 
[Backstage passes to a TAFKAL80ETC concert, 20, 48] => Backstage passes to a TAFKAL80ETC concert, 19, 49 
[Backstage passes to a TAFKAL80ETC concert, 20, 49] => Backstage passes to a TAFKAL80ETC concert, 19, 50 
[Backstage passes to a TAFKAL80ETC concert, 20, 60] => Backstage passes to a TAFKAL80ETC concert, 19, 60 
[Backstage passes to a TAFKAL80ETC concert, 20, 100] => Backstage passes to a TAFKAL80ETC concert, 19, 100 
[Backstage passes to a TAFKAL80ETC concert, 50, 0] => Backstage passes to a TAFKAL80ETC concert, 49, 1 
[Backstage passes to a TAFKAL80ETC concert, 50, -1] => Backstage passes to a TAFKAL80ETC concert, 49, 0 
[Backstage passes to a TAFKAL80ETC concert, 50, 1] => Backstage passes to a TAFKAL80ETC concert, 49, 2 
[Backstage passes to a TAFKAL80ETC concert, 50, 10] => Backstage passes to a TAFKAL80ETC concert, 49, 11 
[Backstage passes to a TAFKAL80ETC concert, 50, 30] => Backstage passes to a TAFKAL80ETC concert, 49, 31 
[Backstage passes to a TAFKAL80ETC concert, 50, 48] => Backstage passes to a TAFKAL80ETC concert, 49, 49 
[Backstage passes to a TAFKAL80ETC concert, 50, 49] => Backstage passes to a TAFKAL80ETC concert, 49, 50 
[Backstage passes to a TAFKAL80ETC concert, 50, 60] => Backstage passes to a TAFKAL80ETC concert, 49, 60 
[Backstage passes to a TAFKAL80ETC concert, 50, 100] => Backstage passes to a TAFKAL80ETC concert, 49, 100 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, -1] => Sulfuras, Hand of Ragnaros, 0, -1 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 30] => Sulfuras, Hand of Ragnaros, 0, 30 
[Sulfuras, Hand of Ragnaros, 0, 48] => Sulfuras, Hand of Ragnaros, 0, 48 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 60] => Sulfuras, Hand of Ragnaros, 0, 60 
[Sulfuras, Hand of Ragnaros, 0, 100] => Sulfuras, Hand of Ragnaros, 0, 100 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, -1] => Sulfuras, Hand of Ragnaros, -1, -1 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[Sulfuras, Hand of Ragnaros, -1, 30] => Sulfuras, Hand of Ragnaros, -1, 30 
[Sulfuras, Hand of Ragnaros, -1, 48] => Sulfuras, Hand of Ragnaros, -1, 48 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 60] => Sulfuras, Hand of Ragnaros, -1, 60 
[Sulfuras, Hand of Ragnaros, -1, 100] => Sulfuras, Hand of Ragnaros, -1, 100 
[Sulfuras, Hand of Ragnaros, 3, 0] => Sulfuras, Hand of Ragnaros, 3, 0 
[Sulfuras, Hand of Ragnaros, 3, -1] => Sulfuras, Hand of Ragnaros, 3, -1 
[Sulfuras, Hand of Ragnaros, 3, 1] => Sulfuras, Hand of Ragnaros, 3, 1 
[Sulfuras, Hand of Ragnaros, 3, 10] => Sulfuras, Hand of Ragnaros, 3, 10 
[Sulfuras, Hand of Ragnaros, 3, 30] => Sulfuras, Hand of Ragnaros, 3, 30 
[Sulfuras, Hand of Ragnaros, 3, 48] => Sulfuras, Hand of Ragnaros, 3, 48 
[Sulfuras, Hand of Ragnaros, 3, 49] => Sulfuras, Hand of Ragnaros, 3, 49 
[Sulfuras, Hand of Ragnaros, 3, 60] => Sulfuras, Hand of Ragnaros, 3, 60 
[Sulfuras, Hand of Ragnaros, 3, 100] => Sulfuras, Hand of Ragnaros, 3, 100 
[Sulfuras, Hand of Ragnaros, 17, 0] => Sulfuras, Hand of Ragnaros, 17, 0 
[Sulfuras, Hand of Ragnaros, 17, -1] => Sulfuras, Hand of Ragnaros, 17, -1 
[Sulfuras, Hand of Ragnaros, 17, 1] => Sulfuras, Hand of Ragnaros, 17, 1 
[Sulfuras, Hand of Ragnaros, 17, 10] => Sulfuras, Hand of Ragnaros, 17, 10 
[Sulfuras, Hand of Ragnaros, 17, 30] => Sulfuras, Hand of Ragnaros, 17, 30 
[Sulfuras, Hand of Ragnaros, 17, 48] => Sulfuras, Hand of Ragnaros, 17, 48 
[Sulfuras, Hand of Ragnaros, 17, 49] => Sulfuras, Hand of Ragnaros, 17, 49 
[Sulfuras, Hand of Ragnaros, 17, 60] => Sulfuras, Hand of Ragnaros, 17, 60 
[Sulfuras, Hand of Ragnaros, 17, 100] => Sulfuras, Hand of Ragnaros, 17, 100 
[Sulfuras, Hand of Ragnaros, 20, 0] => Sulfuras, Hand of Ragnaros, 20, 0 
[Sulfuras, Hand of Ragnaros, 20, -1] => Sulfuras, Hand of Ragnaros, 20, -1 
[Sulfuras, Hand of Ragnaros, 20, 1] => Sulfuras, Hand of Ragnaros, 20, 1 
[Sulfuras, Hand of Ragnaros, 20, 10] => Sulfuras, Hand of Ragnaros, 20, 10 
[Sulfuras, Hand of Ragnaros, 20, 30] => Sulfuras, Hand of Ragnaros, 20, 30 
[Sulfuras, Hand of Ragnaros, 20, 48] => Sulfuras, Hand of Ragnaros, 20, 48 
[Sulfuras, Hand of Ragnaros, 20, 49] => Sulfuras, Hand of Ragnaros, 20, 49 
[Sulfuras, Hand of Ragnaros, 20, 60] => Sulfuras, Hand of Ragnaros, 20, 60 
[Sulfuras, Hand of Ragnaros, 20, 100] => Sulfuras, Hand of Ragnaros, 20, 100 
[Sulfuras, Hand of Ragnaros, 50, 0] => Sulfuras, Hand of Ragnaros, 50, 0 
[Sulfuras, Hand of Ragnaros, 50, -1] => Sulfuras, Hand of Ragnaros, 50, -1 
[Sulfuras, Hand of Ragnaros, 50, 1] => Sulfuras, Hand of Ragnaros, 50, 1 
[Sulfuras, Hand of Ragnaros, 50, 10] => Sulfuras, Hand of Ragnaros, 50, 10 
[Sulfuras, Hand of Ragnaros, 50, 30] => Sulfuras, Hand of Ragnaros, 50, 30 
[Sulfuras, Hand of Ragnaros, 50, 48] => Sulfuras, Hand of Ragnaros, 50, 48 
[Sulfuras, Hand of Ragnaros, 50, 49] => Sulfuras, Hand of Ragnaros, 50, 49 
[Sulfuras, Hand of Ragnaros, 50, 60] => Sulfuras, Hand of Ragnaros, 50, 60 
[Sulfuras, Hand of Ragnaros, 50, 100] => Sulfuras, Hand of Ragnaros, 50, 100 
